hub/policy/src/main/java/uk/gov/ida/hub/policy/redis/SessionStoreRedisCodec.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.ida.hub.policy.redis;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.util.ByteBufferBackedInputStream;
+import com.google.common.io.ByteStreams;
+import io.lettuce.core.codec.RedisCodec;
+import uk.gov.ida.hub.policy.domain.SessionId;
+import uk.gov.ida.hub.policy.domain.State;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+public class SessionStoreRedisCodec implements RedisCodec<SessionId, State> {
+    private final ObjectMapper objectMapper;
+
+    public SessionStoreRedisCodec(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public SessionId decodeKey(ByteBuffer keyBytes) {
+        try {
+            InputStream inputStream = new ByteBufferBackedInputStream(keyBytes);
+            // TODO: Replace with `inputStream.readAllBytes()` when we only use Java 9+
+            return new SessionId(new String(ByteStreams.toByteArray(inputStream)));
```

Comment:
```suggestion
            return new SessionId(new String(ByteStreams.toByteArray(inputStream), StandardCharsets.UTF_8));
```
this is perhaps the nittiest of picks, because there's almost no situation where the locale can screw up a simple hex string, but I instinctively always specify charset just out of habit